### PR TITLE
Fix printing internal libs with the command "dune describe external-lib-deps"

### PR DESCRIPTION
**Pull request by moyodiallo**
When the internal libs are added as dependency by another dune library or dune executable, those are printed.

### CODE REVIEW

**Comment by esope** [comment]
It would really help understanding your fix if you could give an example that first shows the produced output and the expected output, and explains why the expected output should be the right one.
I currently have no idea of what the intended semantics for `external-lib-deps` should be...

**Comment by moyodiallo** [comment]
`dune describe external-lib-deps` is supposed to print only external libraries by dir which contain a dune file.
I simplify the test that I added with more understanding. Feel free to reformulate it.

This PR refers to #6045

**Comment by moyodiallo** [comment]
This was the print before.
```
(default
 ((.
   ((a________ required)
    (inter_lib required)))
  (lib ((a________ required)))))
```
and the right one is
```
(default
 ((. ((a________ required)))
  (lib ((a________ required)))))
```
This is because `inter_lib` is an internal lib.

Refer to the tests that I added.

**Comment by moyodiallo** [comment]
Is it possible to have this PR for dune 3.6 ?
There's a need for https://github.com/ocurrent/opam-dune-lint/pull/46.

@emillon

File: test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/dune-project [hunk 1]
```diff
@@ -0,0 +1,4 @@
+(lang dune 1.9)
+(name foo)
+
+(allow_approximate_merlin true)
```
Comment (emillon) [inline test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/dune-project:4]: can you fix the newline in this file?

File: test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/lib/dune [hunk 2]
```diff
@@ -0,0 +1,3 @@
+(library
+ (name inter_lib)
+ (libraries a________))
```
Comment (emillon) [inline test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/lib/dune:3]: why are tests using names with lots of underscores? would `a` work?

File: test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/run.t [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Expected: The dependency "inter_lib" in ./dune file was printed, the "inter_lib" lib is already
```
Comment (emillon) [inline test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/run.t:1]: Can you rename `fix-bug.t` to something a bit more descriptive?

File: test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/lib/dune (hunk 2 above)
Comment (moyodiallo) [reply test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/lib/dune:3]: to be sure, not having an existent lib.

File: test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/run.t (hunk 3 above)
Comment (moyodiallo) [reply test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/run.t:1]: like `exclude-internal-deps.t` is it better ?

Comment (emillon) [reply test/blackbox-tests/test-cases/external-lib-deps/fix-bug.t/run.t]: yes, that works

**Comment by emillon** [comment]
Thanks!

**Review by emillon — APPROVED** [review]

**Pull request merged**